src/gt4py/next/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -118,7 +118,7 @@ def _process_connectivity_args(
 
         for name, connectivity in offset_provider.items():
             if isinstance(connectivity, Connectivity):
-                if connectivity.index_type not in [np.int32, np.int64]:
+                if connectivity.index_type not in [np.dtypes.Int64DType(), np.dtypes.Int32DType()]:
```

Comment:
this should compare against types defined in `_core.definitions`, but my guess is that the current problems can (maybe should) be fixed on the usage side.